src/ptdf_calculations.jl

```diff
@@ -22,74 +22,35 @@ function binfo_check(binfo::Int)
     return
 end
 
-function _buildptdf(branches, nodes, dist_slack::Vector{Float64})
-    buscount = length(nodes)
-    linecount = length(branches)
+# version 1: use same structure as already present ###########################
 
-    bus_lookup = _make_ax_ref(nodes)
-
-    A = zeros(Float64, buscount, linecount)
-    inv_X = zeros(Float64, linecount, linecount)
-
-    #build incidence matrix
-    for (ix, b) in enumerate(branches)
-        if isa(b, PSY.DCBranch)
-            @warn("PTDF construction ignores DC-Lines")
-            continue
-        end
-
-        (fr_b, to_b) = get_bus_indices(b, bus_lookup)
-        A[fr_b, ix] = 1
-        A[to_b, ix] = -1
-
-        inv_X[ix, ix] = PSY.get_series_susceptance(b)
+function _buildptdf(branches, nodes::Vector{PSY.Bus},
+    dist_slack::Vector{Float64}, linear_solver::String = "Dense")
+    if linear_solver == "KLU"
+        ptdf, a = calculate_PTDF_matrix_KLU(branches, nodes, dist_slack)
+    elseif linear_solver == "Dense"
+        ptdf, a = calculate_PTDF_matrix_DENSE(branches, nodes, dist_slack)
+    elseif linear_solver == "MKLPardiso"
+        error("MKLPardiso still to be implemented")
     end
 
-    slacks =
-        [bus_lookup[PSY.get_number(n)] for n in nodes if PSY.get_bustype(n) == BusTypes.REF]
-    isempty(slacks) && error("System must have a reference bus!")
+    return ptdf, a
```

Comment:
there is no need to return a here anymore

Reply:
we agreed to keep both for the moment